File: server/services/supabase_client.py
from supabase import create_client, Client
import os
from dotenv import load_dotenv

load_dotenv()

SUPABASE_URL = os.getenv("SUPABASE_URL")
SUPABASE_KEY = os.getenv("SUPABASE_SERVICE_KEY")

if not SUPABASE_URL or not SUPABASE_KEY:
	raise ValueError("SUPABASE_URL and SUPABASE_SERVICE_KEY must be set in the environment variables.")

supabase: Client = create_client(SUPABASE_URL, SUPABASE_KEY)

def get_problem_by_match_id(match_id: str):
    match = supabase.table("matches").select("*").eq("match_id", match_id).single().execute().data
    if not match:
        raise Exception("Match not found")

    problem_id = match["problem_id"]
    problem = supabase.table("problems").select("*").eq("id", problem_id).single().execute().data

    return problem

File: client/src/components/client/supabaseclient.js
import { createClient } from '@supabase/supabase-js';

const supabaseUrl = "https://mhzzqnwqfrysmmhwcwdq.supabase.co"
const supabaseKey = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Im1oenpxbndxZnJ5c21taHdjd2RxIiwicm9sZSI6ImFub24iLCJpYXQiOjE3NDYzMzk0NDcsImV4cCI6MjA2MTkxNTQ0N30.O-OdOrR-1TlDQWEQWzX4f9uIT__Lvh23RGHtZ843_kM"
const supabase = createClient(supabaseUrl, supabaseKey);
export default supabase;